src/Compilers/CSharp/Portable/CodeGen/EmitExpression.cs

```diff
@@ -1481,7 +1481,7 @@ private void EmitCallExpression(BoundCall call, UseKind useKind)
                     {
                         // if method is defined in the struct itself it is assumed to be mutating, unless 
                         // it is a member of a readonly struct and is not a constructor
-                        var receiverAddresskind = methodContainingType.IsReadOnly && method.MethodKind != MethodKind.Constructor ?
+                        var receiverAddresskind = IsReadOnlyCall(method, methodContainingType) ?
```

Comment:
>IsReadOnlyCall [](start = 50, length = 14)

Should `IsReadOnlyCall` return the `AddressKind`?

Reply:
I think it is more readable and the code is easier to refactor/reuse when method does one simple task - decides whether the call may mutate "this". Especially if we add more cases where calls are treated as readonly - like if we allow individual "readonly" members. 
It is not a huge overhead. 

---
In reply to: [154444034](https://github.com/dotnet/roslyn/pull/23438#discussion_r154444034) [](ancestors = 154444034)